modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ValueProvidersParameterDeclarationEnricher.java

```diff
@@ -174,6 +190,47 @@ private void enrichParameter(OfValueInformation ofValueInformation,
                                        name, getValueProviderId(ofValueInformation.getValue())));
   }
 
+  private void enrichParameterFields(List<FieldValues> fieldsValues, ParameterDeclaration paramDeclaration,
```

Comment:
I think something can be done to avoid duplication with the code from enrichParameter

Reply:
Was this one addressed?

Reply:
I try but it makes the code awful in my opinion. I can carry on if you think it's that important